src/components/DashboardPanel/DashboardPanel.js

```diff
@@ -28,10 +26,32 @@ export default class DashboardPanel extends Component {
       },
       isStacked: true,
     };
+
+    this.pullData = this.pullData.bind(this);
+  }
+
+  componentWillMount() {
+    this.pullData();
+  }
+
+  /**
+   * Pulls data for all nodes in selectedNodes. Async method, to wait for fetch to finish.
+   */
+  async pullData() {
+    const promiseList = [];
+    this.props.selectedNodes.forEach((node) => {
+      promiseList.push(getNodeUsageData(node, this.state.startDate, this.state.endDate));
```

Comment:
could use object deconstruction here.

`const { startDate, endDate } = this.state;`